From 117489e50c72eac84dbe433cfc6973675957c94c Mon Sep 17 00:00:00 2001
From: Tim Murray <timmurray@google.com>
Date: Mon, 15 Oct 2018 16:50:23 -0700
Subject: [PATCH 48/57] wm: recycle bitmaps immediately in
 TaskSnapshotPersister

Bitmap created in TaskSnapshotPersister are very short lived and
shouldn't be left around in the Java heap.

Test: boot, switch apps, works
bug 117795621

Merged-in: I4b5e0db50c2b7adaa71cb0d22535c1b37c7523e8
Change-Id: I4b5e0db50c2b7adaa71cb0d22535c1b37c7523e8
(cherry picked from commit 9b4eaf740957663b01d4bf056df250dd028e0cb0)
---
 .../core/java/com/android/server/wm/TaskSnapshotPersister.java | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/services/core/java/com/android/server/wm/TaskSnapshotPersister.java b/services/core/java/com/android/server/wm/TaskSnapshotPersister.java
index a642e6ab744..d1c0443c58e 100644
--- a/services/core/java/com/android/server/wm/TaskSnapshotPersister.java
+++ b/services/core/java/com/android/server/wm/TaskSnapshotPersister.java
@@ -360,6 +360,7 @@ class TaskSnapshotPersister {
 
             // For snapshots with reduced resolution, do not create or save full sized bitmaps
             if (mSnapshot.isReducedResolution()) {
+                swBitmap.recycle();
                 return true;
             }
 
@@ -372,6 +373,8 @@ class TaskSnapshotPersister {
                 Slog.e(TAG, "Unable to open " + file + " for persisting.", e);
                 return false;
             }
+            reduced.recycle();
+            swBitmap.recycle();
             return true;
         }
     }
-- 
2.17.1

